tests/components/weheat/test_config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Test the Weheat config flow."""
+
+from unittest.mock import AsyncMock, patch
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.application_credentials import (
+    ClientCredential,
+    async_import_client_credential,
+)
+from homeassistant.components.weheat.const import DOMAIN, OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.setup import async_setup_component
+
+from .const import CLIENT_ID, CLIENT_SECRET, USER_UUID_1
+
+from tests.test_util.aiohttp import AiohttpClientMocker
+from tests.typing import ClientSessionGenerator
+
+
+@pytest.fixture
+async def setup_credentials(hass: HomeAssistant) -> None:
+    """Fixture to setup credentials."""
+    assert await async_setup_component(hass, "application_credentials", {})
+    await async_import_client_credential(
+        hass,
+        DOMAIN,
+        ClientCredential(CLIENT_ID, CLIENT_SECRET),
+    )
+
+
+@pytest.mark.usefixtures("current_request_with_host")
+async def test_full_flow(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+    setup_credentials,
+) -> None:
+    """Check full of adding a single heat pump."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    await handle_oath(hass, hass_client_no_auth, aioclient_mock, result)
+
+    with (
+        patch(
+            "homeassistant.components.weheat.async_setup_entry", return_value=True
+        ) as mock_setup,
+        patch(
+            "homeassistant.components.weheat.config_flow.get_user_id_from_token",
+            new_callable=AsyncMock,
+        ) as mock_weheat,
+    ):
+        mock_weheat.return_value = USER_UUID_1
+        await hass.config_entries.flow.async_configure(result["flow_id"])
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_weheat.mock_calls) == 1
```

Comment:
we should check the result from the async_configuer call. we should check the unique_id set, the title, and stored data